### ISSUE TITLE
Maximum call stack size exceeded when deserializing an entity with grandchildren

### ISSUE THREAD

**Issue by ShockwaVee** [guillaumebriday/json-api-response-converter#3]
Hey,

I'm currently facing the issue where I get an error when deserializing an entity which has nested relationships. At least that's what I think, I might be wrong.

You can see an example at https://codesandbox.io/s/amazing-field-fn322?fontsize=14&hidenavigation=1&theme=dark

The main entity here is the "project" which has "leads" and "offers" inside it. Each offer can have a relationship to "offerItems". 
Now, I think this is what's causing the issue because if you delete each offerItem from the included field, it deserializes it properly and no you don't get any errors.

I have tried deserializing this with jsona and some other libraries and it worked, but I can't use any of the other because of the dependency to Vuex ORM. 

Is this a known issue or is there a problem with the data I am receiving?
Thanks for the help.

**Comment by guillaumebriday** [comment]
Hey!

Thank you for reporting your issue, I need to dive in to be sure what happened under the hood.

Will do it this week I think

**Comment by ShockwaVee** [comment]
Awesome, thanks for the quick response!

**Comment by ShockwaVee** [comment]
Any updates on this?

**Comment by guillaumebriday** [comment]
I'm on it

**Comment by guillaumebriday** [comment]
Hum, I actually reproduce the bug. Your example seems to be very similar as this case in the docs : https://jsonapi.org/format/#document-compound-documents

I need to check it again. Can you eventually open a PR?

Thank you

**Comment by guillaumebriday** [comment]
hey @ShockwaVee 

I think I got it. I tried many things but I think your response is incorrect.


You create an infinite loop for example on line 45 in your code sandbox example.

I think you shouldn't declare the relationship offer for offer_items because it's already declared in the offer to the offer_items on line 118 in this case.

Remove all relationships in offer_items and it will work as excepted.

I will think how to prevent this case in the library though.

Thank you
